mantidproject/mantid
# Fix bugs with row highlighting in reflectometry GUI

This PR fixes the following bugs with row highlighting:
- Green highlighting for completed rows was lost for rows while processing is still in progress.
- Updating the rows' state when deleting workspaces from the ADS was very slow

**To test:**

- Open the `ISIS Reflectometry` interface
- Set the instrument to INTER and investigation id to `1120015` and click Search
- Select all the rows in the search results and click Transfer
- Select the first group in the main table and click Process
- Expand the group. The currently-processing row should be yellow and should turn green when complete. The group will be post-processed when all rows have finished and will also go green.
- When the group is complete, click Autoprocess. The first group should remain green (although there may be a "flicker" where it temporarily resets) and other rows/groups should continue processing in the same way. Note that if a group contains only one row then it will not have postprocessing so will not go green itself.

Refs #23027

*This does not require release notes* because **it fixes regressions since the last release**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
